GenderPayGap.WebUI/Helpers/ControllerHelper.cs

```diff
@@ -56,5 +58,21 @@ public static void Throw404IfFeatureDisabled(FeatureFlag featureFlag)
             }
         }
 
+        public static void ThrowIfUserDoesNotHavePermissionsForGivenOrganisation(ClaimsPrincipal aspDotNetUser, IDataRepository dataRepository, long organisationId)
+        {
+            User gpgUser = GetGpgUserFromAspNetUser(aspDotNetUser, dataRepository);
+
+            UserOrganisation userOrg = gpgUser.UserOrganisations.FirstOrDefault(uo => uo.OrganisationId == organisationId);
+            if (userOrg == null)
+            {
+                throw new IncorrectPermissionsException();
+            }
+
+            if (!userOrg.PINConfirmedDate.HasValue)
+            {
+                // TODO Add sensible error
+            }

```

Comment:
Yeah that's great, thanks for the tips! I was leaving this for review because I didn't know which exceptions made the most sense - this all looks good!